WordPress/src/main/java/org/wordpress/android/ui/comments/CommentAdapter.java

```diff
@@ -296,14 +153,31 @@ public CommentModel getItem(int position) {
 
     @Override
     public long getItemId(int position) {
-        return mComments.get(position).getRemoteCommentId();
+        CommentListItem listItem = getItem(position);
+
+        if (listItem instanceof SubHeader) {
+            return listItem.longId();
+        } else if (listItem instanceof Comment) {
+            return ((Comment) listItem).getComment().getRemoteCommentId();
+        }
+        return -1;
```

Comment:
Not fully sure about the following, but since I happened to have the doubt, sharing below 😄 . If it's a dummy point please just ignore it and do not judge me too hard 😜.

I see SubHeader and Comment are different domains so not immediate to create a unique Id (as I think it's required since we are declaring has stable ids as true). I also see that the position of an element can change in the dataset so we should not use the default implementation that returns position here. This said I'm not super sure that the longId for a SubHeader could not collide with a RemoteCommentId 🤔 ; wonder if we could separate domains like (just an example 🙃 ):

- RemoteCommentId -> long >= 0 (and maybe this is a counter on a site and not site/post(?); so on high traffic sites could be pretty high number)
- NO_ID -> -1
- SubHeader -> long < 10 (like longId * -10 or given a hashcode could be negative in principle, using the timestamp * -10 🤔 )

As said not sure how much the above is near to a real case consideration, so let me know your thoughts 🙇